main.tf

```diff
@@ -136,6 +98,13 @@ resource "aws_iam_role_policy_attachment" "cloudwatchagentserverpolicy" {
   role       = join("", aws_iam_role.main.*.name)
 }
 
+## Managed Policy to allow ssm agent to communicate with SSM Manager
+resource "aws_iam_role_policy_attachment" "ssm" {
+  count      = var.create_instance_iam_role ? 1 : 0
+  policy_arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
```

Comment:
I was looking through the documentation about this policy, there is a lot more there than the ssm session manager https://docs.aws.amazon.com/aws-managed-policy/latest/reference/AmazonSSMManagedInstanceCore.html can we create a more "filtered" policy?

Reply:
I have done some comparisons for the permissions and it seems that those may be the bare minimum to for an instance to communicate with systems manager. Please see a list of full possible permissions ["here"](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awssystemsmanager.html)

Reply:
Hi Patrick, I want to insist that this should be addressed, there are two other statements which are not related to ssm session manager and this is not aligned with the "least privilege" principle, if this requires a bit of work then I'm ok for it to go into the backlog/unreleased, and it must be addressed.